### PR TITLE
Add Flower UI for Celery; use API endpoint for ELB health check

### PR DESCRIPTION
**Pull request by hectcastro**
Installs and sets up Flower to provide a UI for the Celery workers. In addition to the UI, Flower also provides API endpoints. These API endpoints are exposed via Nginx, and pinged by a load balancer to determine worker health. The following DNS records are also created:

- `blue-workers.mmw.azavea.com`
- `green-workers.mmw.azavea.com`

Their only purpose is to provide a way for humans to look at Flower remotely. Access is scoped to the `IPAccess` configuration setting.

Connects #21 and #372

---

**Testing**

- [x] Ensure Flower UI comes up on `worker`
- [x] Run some jobs to see if they populate the Flower UI

**Testing (Optional)**

- [x] Create a new worker AMI based off of this branch
- [x] Spin up a new worker stack to see if all of the new AWS resources have been created
- [x] Ensure Flower UI is only visible via `IPAccess`

### CODE REVIEW

**Comment by mmcfarland** [comment]
:+1: Reprovisioned `worker`, ran some jobs and was able to browse them in the Flower UI.

Did not do Optional Tests.

* think there might be a typo in bullet point 1, Testing (Optional) API -> AMI

**Comment by mmcfarland** [comment]
Also, hat-tip to @jwalgran, this looks like a nice little tool.

File: deployment/ansible/roles/model-my-watershed.celery-worker/templates/nginx-flower.conf.j2 [hunk 1]
```diff
@@ -0,0 +1,37 @@
+server {
+  listen *:80;
+  server_name _;
+
+  access_log /var/log/nginx/flower.access.log logstash_json;
+
+  {% if ['packer'] | is_in(group_names) -%}
+  location /version.txt {
+    alias /srv/version.txt;
```
Comment (kdeloach) [inline deployment/ansible/roles/model-my-watershed.celery-worker/templates/nginx-flower.conf.j2]: Minor, but version.txt on app/tiler shows the long sha1 and version.txt shows the short sha1.

Comment (hectcastro) [inline deployment/ansible/roles/model-my-watershed.celery-worker/templates/nginx-flower.conf.j2]: This appears to have been an inconsistency in the SHA produced by `git describe`, which gets used when `GIT_COMMIT` is not supplied. Should be resolved in https://github.com/WikiWatershed/model-my-watershed/commit/f66a4f300d41ac4c8cfc394b430aed168c350461.

**Comment by kdeloach** [comment]
Unrelated to this PR, but while testing, I deleted a worker stack in CloudFormation then ran the launch-stacks command and got this error:

```
...
[DEBUG] Worker-0f217d4f07557999b7479bd9c DELETE_IN_PROGRESS
[DEBUG] Worker-0f217d4f07557999b7479bd9c DELETE_IN_PROGRESS
[DEBUG] Worker-0f217d4f07557999b7479bd9c DELETE_IN_PROGRESS
Traceback (most recent call last):
  File "./mmw_stack.py", line 82, in <module>
    main()
  File "./mmw_stack.py", line 79, in main
    args.func(mmw_config=mmw_config, **vars(args))
  File "./mmw_stack.py", line 16, in launch_stacks
    build_stacks(mmw_config, aws_profile, **kwargs)
  File "/var/projects/model-my-watershed/deployment/cfn/stacks.py", line 82, in build_stacks
    worker_graph.go()
  File "/var/projects/model-my-watershed/env/local/lib/python2.7/site-packages/majorkirby/majorkirby.py", line 289, in go
    self.heartbeat(n)
  File "/var/projects/model-my-watershed/env/local/lib/python2.7/site-packages/majorkirby/majorkirby.py", line 405, in heartbeat
    status = self._check_cfn()
  File "/var/projects/model-my-watershed/env/local/lib/python2.7/site-packages/majorkirby/majorkirby.py", line 353, in _check_cfn
    self.stack = self.boto_conn.describe_stacks(self.stack_name)[0]
  File "/var/projects/model-my-watershed/env/local/lib/python2.7/site-packages/boto/cloudformation/connection.py", line 690, in describe_stacks
    return self.get_list('DescribeStacks', params, [('member', Stack)])
  File "/var/projects/model-my-watershed/env/local/lib/python2.7/site-packages/boto/connection.py", line 1186, in get_list
    raise self.ResponseError(response.status, response.reason, body)
boto.exception.BotoServerError: BotoServerError: 400 Bad Request
<ErrorResponse xmlns="http://cloudformation.amazonaws.com/doc/2010-05-15/">
  <Error>
    <Type>Sender</Type>
    <Code>ValidationError</Code>
    <Message>Stack with id Worker-0f217d4f07557999b7479bd9c does not exist</Message>
  </Error>
  <RequestId>19fda6b1-1b52-11e5-a4bf-93c90a53b567</RequestId>
</ErrorResponse>
```

Don't think it's a real issue but figured it's worth mentioning.

**Comment by hectcastro** [comment]
I think that if you wait for it to finish deleting and then run `launch-stacks`, it'll bring a new worker stack up. Is that what you expected it to do?

**Comment by kdeloach** [comment]
I was just curious to see what would happen if I ran this command while a delete was in progress. It works exactly as I expected. Nothing to do for this issue.

**Comment by kdeloach** [comment]
+1 Tested and verified all of the above.

**Comment by kdeloach** [comment]
This will be insanely helpful for debugging celery workers. The Flower UI is great.

**Pull request merged**